dot/rpc/modules/chain_test.go

```diff
@@ -15,3 +15,473 @@
 // along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
 
 package modules
+
+import (
+	"errors"
+	"math/big"
+	"net/http"
+	"testing"
+
+	apimocks "github.com/ChainSafe/gossamer/dot/rpc/modules/mocks"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChainModule_GetBlock(t *testing.T) {
+	mockedHash := common.NewHash([]byte{0x01, 0x02})
+	emptyBlock := types.NewEmptyBlock()
+
+	mockBlockAPI := new(apimocks.BlockAPI)
+	mockBlockAPI.On("GetBlockByHash", mock.AnythingOfType("common.Hash")).Return(&emptyBlock, nil)
+	mockBlockAPI.On("BestBlockHash").Return(mockedHash, nil)
+
+	mockBlockAPIGetHashErr := new(apimocks.BlockAPI)
+	mockBlockAPIGetHashErr.On("GetBlockByHash", mock.AnythingOfType("common.Hash")).Return(nil, errors.New("GetJustification error"))
+
+	bodyBlock := types.NewEmptyBlock()
+	bodyBlock.Body = types.BytesArrayToExtrinsics([][]byte{{1}})
+	mockBlockAPIWithBody := new(apimocks.BlockAPI)
+	mockBlockAPIWithBody.On("GetBlockByHash", mock.AnythingOfType("common.Hash")).Return(&bodyBlock, nil)
+
+	chainModule := NewChainModule(mockBlockAPI)
+	res := ChainBlockResponse{}
+	type fields struct {
+		blockAPI BlockAPI
+	}
+	type args struct {
+		r   *http.Request
+		req *ChainHashRequest
+		res *ChainBlockResponse
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
```

Comment:
You should have an actual error here instead. I've seen cases where the wrong error was returned so asserting the actual error message is useful.

And then you can assert it with something like

```go
if testCase.err != nil {
	assert.EqualError(t, err, testCase.err.Error())
} else {
	assert.NoError(t, err)
}
```

Reply:
If there's not too many error cases being produced by the function I don't mind using `wantErr`.

Reply:
Yes it makes sense